src/uniq.js

```diff
@@ -0,0 +1,74 @@
+var common = require('./common');
+var fs = require('fs');
+
+//add c spaces to the left of str
+function lpad(c, str){
+    var res = "" + str;
+    if(res.length < c){
+        res = Array((c-res.length)+1).join(" ") + res;
+    }
+    return res;
+}
+
+//@
+//@ ### uniq([options,] [input, [output]])
+//@ Available options:
+//@
+//@ + `-i`: Ignore differences in case when comparing
+//@ + `-c`: Prefix lines by the number of occurrences
+//@ + `-d`: Only print duplicate lines, one for each group
+//@
+//@ Examples:
+//@
+//@ ```javascript
+//@ uniq('foo.txt');
+//@ uniq('-i', 'foo.txt');
+//@ uniq('-cd', 'foo.txt', 'bar.txt');
+//@ ```
+//@
+//@ Filter adjacent matching lines from input
+function _uniq(options, input, output) {
+  options = common.parseOptions(options, {
+    'i': 'ignoreCase',
+    'c': 'count',
+    'd': 'duplicates'
+  });
+
+  // Check if this is coming from a pipe
+  var pipe = common.readFromPipe(this);
+
+  if (!input && !pipe)
+    common.error('no input given');
+
+  var lines = (input ? fs.readFileSync(input, 'utf8') : pipe).
+              trimRight().
+              split(/\r*\n/);
+
+  //Perform a run-length encoding of the lines
+  var uniqed = [{count: 1, ln: lines[0]}];
+  lines.slice(1).forEach(function(line){
+      var cmp = options.ignoreCase ? 
```

Comment:
Could you please update this to 2 spaces for indentation? Looks like it's 4 here. Thanks!